src/Functions/S2CapContains.cpp

```diff
@@ -0,0 +1,121 @@
+#include "config_functions.h"
+
+#include <Columns/ColumnsNumber.h>
+#include <Columns/ColumnTuple.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/IFunctionImpl.h>
+#include <Common/typeid_cast.h>
+#include <ext/range.h>
+
+#include "../contrib/s2geometry/src/s2/s2latlng.h"
+#include "../contrib/s2geometry/src/s2/s2cell_id.h"
+#include "../contrib/s2geometry/src/s2/s2cap.h"
+#include "../contrib/s2geometry/src/s2/s1angle.h"
+
+namespace DB 
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+namespace
+{
+
+/// TODO: Comment this
+class FunctionS2CapContains : public IFunction
+{
+public:
+    static constexpr auto name = "S2CapContains";
+
+    static FunctionPtr create(ContextPtr)
+    {
+        return std::make_shared<FunctionS2CapContains>();
+    }
+
+    std::string getName() const override
+    {
+        return name;
+    }
+
+    size_t getNumberOfArguments() const override { return 3; }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        size_t number_of_arguments = arguments.size();
+
+        if (number_of_arguments != 3) {
+            throw Exception("Number of arguments for function " + getName() + " doesn't match: passed "
+                + toString(number_of_arguments) + ", should be 3",
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+        }
+
+        const auto * arg = arguments[0].get();
+
+        if (!WhichDataType(arg).isUInt64()) {
+            throw Exception(
+                "Illegal type " + arg->getName() + " of argument " + std::to_string(1) + " of function " + getName() + ". Must be UInt64",
```

Comment:
std::to_string(1) change to "1", it is redundant, same for other arguments

Reply:
Also there is another constructor for Exception with fmt-like args, i.e.

```c
throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Illegal type {} of argument 1 of functoin. Must be UInt64.", arg->getName(), getName());
```